gnbsim.go

```diff
@@ -67,38 +69,50 @@ func action(c *cli.Context) error {
 		return err
 	}
 
-	summaryChan := make(chan common.InterfaceMessage)
 	result := "PASS"
 
+	var profileWaitGrp sync.WaitGroup
 	for _, profileCtx := range config.Configuration.Profiles {
 		if !profileCtx.Enable {
 			continue
 		}
-		logger.AppLog.Infoln("executing profile:", profileCtx.Name,
-			", profile type:", profileCtx.ProfileType)
+		profileWaitGrp.Add(1)
 
-		go profile.ExecuteProfile(profileCtx, summaryChan)
+		go func(profileCtx *profctx.Profile) {
+			defer profileWaitGrp.Done()
+			profileCtx.SummaryChan = make(chan common.InterfaceMessage)
+			logger.AppLog.Infoln("executing profile:", profileCtx.Name,
+				", profile type:", profileCtx.ProfileType)
 
-		// Waiting for execution summary from profile routine
-		msg, ok := (<-summaryChan).(*common.SummaryMessage)
-		if !ok {
-			logger.AppLog.Fatalln("Invalid Message Type")
-		}
+			go profile.ExecuteProfile(profileCtx)
 
-		logger.AppSummaryLog.Infoln("Profile Name:", msg.ProfileName, ", Profile Type:", msg.ProfileType)
-		logger.AppSummaryLog.Infoln("Ue's Passed:", msg.UePassedCount, ", Ue's Failed:", msg.UeFailedCount)
+			// Waiting for execution summary from profile routine
+			msg, ok := (<-profileCtx.SummaryChan).(*common.SummaryMessage)
+			if !ok {
+				logger.AppLog.Fatalln("Invalid Message Type")
+			}
 
-		if msg.UeFailedCount != 0 {
-			result = "FAIL"
-		}
+			logger.AppSummaryLog.Infoln("Profile Name:", msg.ProfileName, ", Profile Type:", msg.ProfileType)
+			logger.AppSummaryLog.Infoln("Ue's Passed:", msg.UePassedCount, ", Ue's Failed:", msg.UeFailedCount)
 
-		if len(msg.ErrorList) != 0 {
-			logger.AppSummaryLog.Infoln("Profile Errors:")
-			for _, err := range msg.ErrorList {
-				logger.AppSummaryLog.Errorln(err)
+			if msg.UeFailedCount != 0 {
+				result = "FAIL"
 			}
+
+			if len(msg.ErrorList) != 0 {
+				logger.AppSummaryLog.Infoln("Profile Errors:")
+				for _, err := range msg.ErrorList {
+					logger.AppSummaryLog.Errorln(err)
+				}
+			}
+		}(profileCtx)
+		if config.Configuration.ExecInParallel == false {
+			profileWaitGrp.Wait()
 		}
 	}
+	if config.Configuration.ExecInParallel == true {
+		profileWaitGrp.Wait()
```

Comment:
As a solution, Can we not use a single summary channel as it was previously used keeping everything same.
So instead of using Wait() call to block at line 110 and 114, we can wait on the common summary channel,
this will log summary details for one profile at a time